### PR TITLE
TST: add tests for stats.tvar with unflattened arrays

### PR DESCRIPTION
**Pull request by clementkng**
Adds tests to gh-8359. Also a cleanup of the previous [PR](https://github.com/scipy/scipy/pull/10264).

### CODE REVIEW

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -77,6 +77,19 @@ def test_tvar(self):
         y = stats.tvar(X, limits=None)
         assert_approx_equal(y, X.var(ddof=1), significant=self.dprec)
 
+        X_UNFLATTENED = arange(63, dtype=float64).reshape((9, 7))
```
Comment (rgommers) [inline scipy/stats/tests/test_stats.py]: last request (probably): can you change this name to `x` or `x_2d`? better to write modern-looking PEP8-compliant code:)

**Review by rgommers — COMMENTED** [review]
There's still the issue that `np.ma.var` returns a masked array unless `axis` is None:
```
In [20]: np.ma.var(x, axis=0)
Out[20]: 
masked_array(data=[326.66666667, 326.66666667, 326.66666667, 326.66666667,
                   326.66666667, 326.66666667, 326.66666667],
             mask=False,
       fill_value=1e+20)
```

This I consider a bug. This bug wasn't visible before, because for a ravel'ed array `np.ma.var` returns a scalar. So it still needs another tweak to go back from a masked array to an ndarray, and a new set of tests.

On the other hand, maybe masked array is necessary because when `limits` are set, it's possible for all values along a given axis in one position to be masked. So what to do then - return `nan` rather that a masked value perhaps?

**Comment by clementkng** [comment]
I think it personally makes sense, given multiple axes, that a fully masked out axis should return the value of `nan`. Perhaps masked arrays should only be used internally w/i the `tvar()` function?

**Comment by clementkng** [comment]
I think it personally makes sense, given multiple axes, that a fully masked out axis should return the value of `nan`. Perhaps masked arrays should only be used internally w/i the `tvar()` function?

**Comment by rgommers** [comment]
> I think it personally makes sense, given multiple axes, that a fully masked out axis should return the value of nan. Perhaps masked arrays should only be used internally w/i the tvar() function?

Yes, I agree. Let's go with that. Would you be able to implement that in this PR?

**Comment by clementkng** [comment]
Just so I'm clear, I'm going to add a test that fully masks out an axis, and make sure that it returns `nan`. I'm also going to write a test that has some set `limits` and make sure it does not return a masked array.

**Comment by rgommers** [comment]
Yes indeed. Thanks @clementkng

**Comment by clementkng** [comment]
I'm trying to implement the correct behavior by replacing the masked values w/ `nan`s, then using [`np.nanvar`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanvar.html) to do the variance calculation. However, I get `RuntimeWarning: Degrees of freedom <= 0 for slice.`, necessarily b/c I'm constructing a test case where all the values are fully masked out. Any tips for suppressing the warning? Or is this desired behavior?

**Comment by rgommers** [comment]
> Any tips for suppressing the warning? Or is this desired behavior?

No, not desired. Inside the tests you can suppress warnings with `with suppress_warnings ...:`, e.g., https://github.com/scipy/scipy/blob/master/scipy/stats/tests/test_stats.py#L102

**Comment by clementkng** [comment]
@rgommers I merged to pick up the new `pybind11` package, but I'm not sure how to fix those `Exit code 139` errors in the Azure Linux builds. From what I've Googled, it seems to be a Linux program crash. Should I rebase and try again?

**Comment by rgommers** [comment]
>  I merged to pick up the new pybind11 package, but I'm not sure how to fix those Exit code 139 errors in the Azure Linux builds. From what I've Googled, it seems to be a Linux program crash. Should I rebase and try again?

That must be unrelated to this PR, don't worry about it for now. If your tests pass, then this is good to go. 

I restarted the crashed build to see if it goes away.

**Comment by clementkng** [comment]
@rgommers Thanks for restarting the build. Is there any additional feedback you'd like for me to address?

**Comment by rgommers** [comment]
LGTM now and all green, so in it goes. Thanks @clementkng

**Pull request merged**